Sources/ProjectDescription/Workspace.swift

```diff
@@ -18,6 +38,9 @@ public struct Workspace: Codable, Equatable {
     /// List of files to include in the workspace (e.g. Documentation)
     public let additionalFiles: [FileElement]
 
+    /// Generation options.
+    public let generationOptions: [GenerationOptions]
```

Comment:
I do agree that the array semantics make it awkward. We could solve that in part with a `Set`, but that still feels like a half measure. I think improving the API where it can be is a wise choice, so we don't get locked into supporting something subpar. I'm game to change it to a struct, if you'd like.

Reply:
I implemented the `struct` solution and was able to delete so much code :)